jib-core/src/main/java/com/google/cloud/tools/jib/builder/BuildConfiguration.java

```diff
@@ -226,44 +210,50 @@ public static boolean isValidJavaClass(String className) {
    * @throws NumberFormatException if any of the ports are in an invalid format or out of range
    */
   @VisibleForTesting
-  static ImmutableSortedMap<String, EmptyStruct> portListToPortMap(List<String> ports)
-      throws NumberFormatException {
-    SortedMap<String, EmptyStruct> result = new TreeMap<>();
+  static ImmutableList<String> expandPortRanges(List<String> ports, BuildLogger logger) {
+    List<String> result = new ArrayList<>();
 
     for (String port : ports) {
       // Remove all spaces
-      port = port.trim().replace(" ", "");
+      port = port.replace(" ", "");
```

Comment:
I don't think we should be allowing the user to type like 1_2____4_3 and have it be a valid port.